rest/src/main/java/com/redhat/ipaas/rest/v1/handler/integration/IntegrationHandler.java

```diff
@@ -95,6 +92,17 @@ public Integration create(Integration integration) {
         return Creator.super.create(updatedIntegration);
     }
 
+    private void updateStatus(String id, Integration.Status desiredStatus) {
+        this.update(id, new Integration.Builder()
+            .createFrom(get(id))
+            .desiredStatus(desiredStatus)
+            // Set the current status to 'pending' immediately when
+            // a status change is requested.
+            // This status will be later changed by the activation handlers.
+            .currentStatus(Integration.Status.Pending)
```

Comment:
Yeah, don't like the imperative style, too (as it's not the callers duty to do state changes)

let me think about it when I pick up changes from #349